### DISCUSSION TITLE
Inspect surrounding blocks with Robots

### DISCUSSION THREAD

**Discussion by Occhioverde** [fnuecke/oc2#282] [Q&A] [answered]
Hi all!
Today I discovered this mod and tried it for the first time to build some automations, but noticed that the Robots cannot inspect the surrounding environment by returning info about one of the six block around them (id, metadata, etc.), which would be useful to improve the interactions with the world.
I guess that I can implement this functionality with a custom module, but I first wanted to ask if there was a simpler and built-in way to achieve this or whether someone already implemented it.
Thank you in advance!

**Comment by CoolCat467** [comment, marked as answer]
There was some discussion about this in https://github.com/fnuecke/oc2/pull/108, but nothing's happened in quite some time.

**Reply by Occhioverde** [reply]
Thank you very much! I searched for this feature in the documentation and for requests in the issues, but completely forgot to look at the pull requests.
It's a real shame that this project is currently halted and that it wasn't merged, but at least I have something to work on to implement it myself.

**Reply by CoolCat467** [reply]
You are welcome!
You've probably seen, but apparently the main reason development has halted is because there is some sort of underlying issue in Sedna, the Java-based RISC-V emulator.
Best of luck to you!
